src/Microsoft.OData.Core/ODataBatchReaderStream.cs

```diff
@@ -10,129 +10,51 @@ namespace Microsoft.OData
     using System;
     using System.Collections.Generic;
     using System.Diagnostics;
+    using System.IO;
     using System.Text;
     #endregion Namespaces
 
     /// <summary>
     /// Class used by the <see cref="ODataBatchReader"/> to read the various pieces of a batch payload.
     /// </summary>
     /// <remarks>
-    /// This stream separates a batch payload into multiple parts by scanning ahead and matching
-    /// a boundary string against the current payload.
+    /// This is the base class for format-specific derived classes to process batch requests in different formats,
+    /// such as multipart/mixed format and application/json format.
     /// </remarks>
-    internal sealed class ODataBatchReaderStream
+    internal abstract class ODataBatchReaderStream
     {
-        /// <summary>
-        /// The default length for the line buffer byte array used to read lines; expecting lines to normally be less than 2000 bytes.
-        /// </summary>
-        private const int LineBufferLength = 2000;
-
-        /// <summary>
-        /// The byte array used for reading lines from the stream. We cache the byte array on the stream instance
-        /// rather than allocating a new one for each ReadLine call.
-        /// </summary>
-        private readonly byte[] lineBuffer;
-
-        /// <summary>The input context to read the content from.</summary>
-        private readonly ODataRawInputContext inputContext;
-
-        /// <summary>The boundary string for the batch structure itself.</summary>
-        private readonly string batchBoundary;
-
         /// <summary>The buffer used by the batch reader stream to scan for boundary strings.</summary>
-        private readonly ODataBatchReaderStreamBuffer batchBuffer;
-
-        /// <summary>The media type resolver to use when interpreting the incoming content type.</summary>
-        private readonly ODataMediaTypeResolver mediaTypeResolver;
+        protected readonly ODataBatchReaderStreamBuffer BatchBuffer;

```

Comment:
>BatchBuffer [](start = 56, length = 11)

I see only this variable is renamed to "B"atchBuffer from  batchBuffer. But other variable are still camel case.  I would suggest  to pick and follow one standard

Reply:
StyleCop gives a error about non-private variable should not be in camel case, which should be used for private variable.

Reply:
Actually, standard build requires that Readonly variables that are not declared private must start with an upper-case letter, otherwise there will be a warning. 

---
In reply to: [140608066](https://github.com/OData/odata.net/pull/960#discussion_r140608066) [](ancestors = 140608066)